### PR TITLE
Add base HistoryActivity for testing

### PR DESCRIPTION
**Pull request by danesfeder**
This PR extracts the history code from `ComponentNavigationActivity` into a base `HistoryActivity` class so we can more easily add it to our examples for testing purposes.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,6 +18,7 @@
             <intent-filter>
                 <action android:name="android.intent.action.MAIN"/>
                 <category android:name="android.intent.category.LAUNCHER"/>
+                <action android:name="android.intent.action.VIEW"/>
```
Comment (Guardiola31337) [inline app/src/main/AndroidManifest.xml]: Curious 🤔 Why is this needed?

Comment (danesfeder) [reply app/src/main/AndroidManifest.xml]: screen shot 2018-10-23 at 9 33 21 am

This lint warning was highlighting the whole `AndroidManifest.xml`
